### PR TITLE
Add "Clear Non-existent Files" to Open Recent menu. Close #1457

### LINKED ISSUE #1457: Add entry to remove all non-existent files from File -> Open Recent menu

### PR DESCRIPTION
**Pull request by Idloj**

### CODE REVIEW

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 1]
```diff
@@ -1292,14 +1298,28 @@ void MainWindow::slot_open_recent()
...
     settings.setValue("recent_file_list", QStringList());
 
     update_recent_files_menu(QStringList());
 }
 
+void MainWindow::slot_clear_recent_files_nonexistent()
+{
+    QSettings settings(SETTINGS_ORGANIZATION, SETTINGS_APPLICATION);
+    QStringList files = settings.value("recent_file_list").toStringList();
+    QStringList existentFiles;
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: We use snake_case for variables (detailed coding guidelines: https://github.com/appleseedhq/appleseed/wiki/Coding-Philosophy-and-Guidelines).

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 2]
```diff
@@ -1292,14 +1298,28 @@ void MainWindow::slot_open_recent()
...
+
+    for (int i = 0; i < files.size(); i++)
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: Please add curly braces around multi-lines `for` loop body.

File: src/appleseed.studio/mainwindow/mainwindow.cpp (hunk 1 above)
Comment (Idloj) [reply src/appleseed.studio/mainwindow/mainwindow.cpp]: Oh right, I forgot

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 3]
```diff
@@ -1292,14 +1297,30 @@ void MainWindow::slot_open_recent()
...
 
+void MainWindow::slot_clear_recent_nonexistent_project_files()
+{
+    QSettings settings(SETTINGS_ORGANIZATION, SETTINGS_APPLICATION);
+    QStringList files = settings.value("recent_file_list").toStringList();
+    QStringList existent_files;
+
+    for (int i = 0; i < files.size(); i++)
+    {
+        if (QFileInfo(files[i]).isFile())
```
Comment (Idloj) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: I made sure `isFile` is appropriate for what we want: it return false for non-existent files and for directories.

**Pull request merged**